### ISSUE TITLE
Earlystopping and predicting mode?

### ISSUE THREAD

**Issue by dalcsy** [iyhaoo/DISC#6]
Hi,

I'm recently using the DISC algorithm on my scRNA dataset. I found that after several rounds, the loss no longer decreased, but the algorithm is still running for tens of rounds. Is there someway for earlystopping?
And, if I do not want to train the data, but just predict it with the already built model saved in .pb file, what should I do?

Thank you!

**Comment by iyhaoo** [comment]
Hi!
Thank you very much for using DISC!
Yes, you can change parameters "--warm-up-cells", "--round-number" and "--training-round-size" to make it stop earlier. The "--warm-up-cells" specifies at least how many cells will input to the model (please note that cells can be used for more than once if your dataset doesn't have so many cells, for example, the default setting is 5 million and if your dataset have only 10 thousand cells then each cell will be repeatly trained by 100 times). Then "--round-number" and "--training-round-size" together decide the window after warming up (--warm-up-cells). In this window, if the model reaches its "optimal point" (define by ourselves based on some metrics), then it will stop and give the model of the "optimal point" as our final model, then DISC does inference.
If you want to do inference based on your pretrained DISC model, you can adjust your code based on the end of "https://github.com/iyhaoo/DISC/blob/master/reproducibility/Data%20Preparation%2C%20Imputation%20and%20Computational%20Resource%20Evaluation/Run%20Imputation.md".
Enjoy the evening (with DISC hahaha)!
Best regards,
Hao
